src/js/components/dropdown.js

```diff
@@ -0,0 +1,286 @@
+    this.menuElement = this.element.querySelector(this.menuAttribute);
+
+    // Keeps track of the currently active dropdown and helps with focus management
+    this.activeDropdown = null;
+
+    // Bind methods
+    this._handleClick = this._handleClick.bind(this);
+    this._handleKeydown = this._handleKeydown.bind(this);
+
+    this.init();
+  }
+
+  /**
+   * Opens a dropdown menu
+   * @param {HTMLButtonElement} toggle - a button element which toggles the dropdown
+   * @param {HTMLDivElement} menu - a div containing the dropdown menu items
+   */
+  openMenu(toggle, menu) {
+    const toggleButton = toggle || this.toggleElement;
+    const menuList = menu || this.menuElement;
+
+    // Return if disabled dropdown is being opened programmatically
+    if (toggleButton.hasAttribute('disabled')) {
+      return;
+    }
+
+    // If the menu was opened by clicking an associated toggle
+    if (toggleButton && toggleButton !== null) {
+      toggleButton.setAttribute('aria-expanded', 'true');
+    }
+
+    // Remove the 'hidden' attribute to show the menu
+    menuList.setAttribute('aria-hidden', 'false');
+
+    this.activeDropdown = this.element;
+  }
+
+  /**
+   * Closes a dropdown menu
+   * @param {HTMLButtonElement} toggle - a button element which toggles the dropdown
+   * @param {HTMLDivElement} menu - a div containing the dropdown menu items
+   */
+  closeMenu(toggle, menu) {
+    const toggleButton = toggle || this.toggleElement;
+    const menuList = menu || this.menuElement;
+
+    // Return if disabled dropdown is being closed programmatically
+    if (toggleButton.hasAttribute('disabled')) {
+      return;
+    }
+
+    if (toggleButton && toggleButton !== undefined) {
+      toggleButton.setAttribute('aria-expanded', 'false');
+    }
+
+    menuList.setAttribute('aria-hidden', 'true');
+  }
+
+  /**
+   * Creates an object of menu items within a dropdown
+   * @param {HTMLDivElement} menu - a div containing the dropdown menu items
+   * @returns {Object} An object of menu item anchor elements
+   */
+  _setUpMenu(menu) {
+    const menuObject = {};
+
+    // Create a real Array of all the focusable elements in the menu
+    const menuFocusables = Array.prototype.slice.call(
+      menu.querySelectorAll(this.focusableElements)
+    );
+
+    // Create a property to hold an array of all focusables
+    menuObject.all = menuFocusables;
+
+    // Create a property with a reference to the first focusable
+    menuObject.first = menuFocusables[0];
+
+    // Create a property with a reference to the last focusable
+    menuObject.last = menuFocusables[menuFocusables.length - 1];
+
+    return menuObject;
+  }
+
+  _handleClick(event) {
+    const dropdown = event.target.closest(this.dropdownAttribute);
+
+    // If click didn't come from within dropdown component, close all open menus
+    if (!dropdown) {
+      this.closeMenu();
+      return;
+    }
+
+    const toggleButton = dropdown.querySelector(this.toggleAttribute);
```

Comment:
Same as comments above, you shouldn't need to reassign these here. You can reference the instance properties for toggle and menu instead.